The Game/Assets/Scripts/Agent/Enemy/Enemy.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Jincom.Core;
+using Jincom.PickUps;
+
+namespace Jincom.Agent
+{
+    public class Enemy
```

Comment:
This is a good idea, remove the stuff about state and keep that in enemy agent base, then also add this Enemy class in enemy agent base